op-challenger/fault/types.go

```diff
@@ -19,19 +19,32 @@ type TraceProvider interface {
 	Get(i uint64) (common.Hash, error)
 }
 
-type Claim struct {
+// ClaimData is the core of a claim. It must be unique inside a specific game.
+
+type ClaimData struct {
 	Value common.Hash
 	Position
 }
 
-type Response struct {
-	Attack bool // note: can we flip this to true == going right / defending??
-	Value  common.Hash
-	Parent Claim
+// Claim extends ClaimData with information about the relationship between two claims.
+// It uses ClaimData to break cyclicity without using pointers.
+// If the position of the game is Depth 0, IndexAtDepth 0 it is the root claim
+// and the Parent field is empty & meaningless.
+type Claim struct {
+	ClaimData
+	Parent ClaimData
+}
+
+func (c *Claim) IsRoot() bool {
+	return c.Position.IsRootPosition()
+}
+
+func (c *Claim) DefendsParent() bool {
+	return (c.IndexAtDepth() >> 1) != c.Parent.IndexAtDepth()
 }
```

Comment:
NBD but having a docstring here would help make clear what defends means for a given claim

Reply:
added